### DISCUSSION TITLE
Need help to run EdgeX using Docker (operation not permitted error)

### DISCUSSION THREAD

**Discussion by hzqhpz-universlink** [edgexfoundry/community#195] [Q&A] [answered]
I keep getting "operation not permitted" for some services that I wanted to run. Thus they kept restarting. I have been running these on Linux Ubuntu 22.04 amd64. I have used kamakura, jakarta and levski.

Apparently, it working just fine in my arm64 device when I run the arm64 version.

![image](https://github.com/edgexfoundry/community/assets/148033793/a8b2dd38-a6ae-4125-bdcd-3eff82e15a92)

**Comment by bnevis-i** [comment]
@hzqhpz-universlink You are trying to run the arm64 containers on x86_64.  (look at the container names).

**Reply by hzqhpz-universlink** [reply]
My apologies for providing the wrong picture. The above is what I'm trying to run arm64 on my machine instead of the default x86_64. This new picture is when I ran the x86_64 version.
![image](https://github.com/edgexfoundry/community/assets/148033793/266f7c38-2c0c-4775-8397-dcb08e4924d9)

**Reply by bnevis-i** [reply]
I am somewhat at a loss.  You seem to be running the no-security version.  There should be very little stopping the containers from starting, especially the 3rd party ones like consul, redis.  Portainer somehow still running.  Do you have enough free disk space?  Start clean with "docker system prune"?

**Reply by hzqhpz-universlink** [reply]
I have enough free disk space and already ran the mentioned command. Still can't do it. The Portainer is actually executed separately, not with the other docker-compose.

**Comment by farshidtz** [comment, marked as answer]
I assume you have installed the Docker engine with the [snap](https://snapcraft.io/docker) and hitting a known issue.

See:
https://github.com/docker-snap/docker-snap#usage
https://bugs.launchpad.net/snapd/+bug/1908448

As a workaround, remove all instances of the following from your docker compose file:
```yaml
    security_opt:
    - no-new-privileges:true
```
I don't know why that option is set and the implications of removing it. @bnevis-i any idea?

**Reply by bnevis-i** [reply]
The `no-new-privileges` option prevents SUID binaries from executing in the container's context.  It prevents a non-root process from re-acquiring root privilege.  It is a defense-in-depth option.

**Reply by hzqhpz-universlink** [reply]
Apparently true I installed Docker using Snap. I reinstalled them using APT, and I can now run EdgeX with no issue. Thank you for the help.

**Comment by cloudxxx8** [comment]
closing this discussion since the problem is resolved.
